webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/users/UserRepository.scala

```diff
@@ -693,18 +709,58 @@ class JdbcUserRepository(doobie: Doobie) extends UserRepository {
                     EXCLUDED.lastlogin,
                     EXCLUDED.statushistory
                   )
-                  """
+                returning id
+          """
+        }
 
-          Update[UserInfo](sql).updateMany(updated)
+        def update(updated: Vector[UserInfo]): ConnectionIO[Vector[String]] = {
+          isCaseSensitive match {
+            case false =>
+              val clause = { // Requires (UserInfo, UserId) parameters.
+                // Deleted users should still be updatable. This will still end with the ON CONFLICT clause.
+                s"""
+                SELECT ?,?,?,?,?,?,?,?,?
+                WHERE NOT EXISTS (
+                  SELECT 1 from users WHERE LOWER(id) = LOWER(?) AND status <> '${UserStatus.Deleted.value}'
+                )
+                """
+              }
+              Update[(UserInfo, String)](updateQuery(clause))
+                .updateManyWithGeneratedKeys[String]("id")(updated.map(v => (v, v.id)))
+                .compile
+                .toVector
+            case true  =>
+              val clause = {
+                """
+                VALUES (?,?,?,?,?,?,?,?,?)
+                """.stripMargin
+              }
+              Update[UserInfo](updateQuery(clause))
+                .updateManyWithGeneratedKeys[String]("id")(updated)
+                .compile
+                .toVector
+          }
         }
 
-        transactIOResult("Erreur when updating the list of") { xa =>
+        transactIOResult("Error when updating the list of users") { xa =>
           (for {
             maybeUpdate       <- updatable.query[UserInfo].to[Vector]
             (zombies, managed) = maybeUpdate.partition(_.status == UserStatus.Deleted)
             updated            = UserRepository.computeUpdatedUserList(users, origin, trace, toMap(zombies), toMap(managed))
-            _                 <- update(updated.values.toVector)
-          } yield ()).transact(xa)
+            dbUpdated         <- update(updated.values.toVector)
+            // compute users that may have not been inserted nor updated because of the case-sensitivity clause in the query
+            diff               = updated.keySet.diff(dbUpdated.toSet)
+          } yield diff).transact(xa).flatMap { nonUpdated =>
+            ZIO
```

Comment:
this bit is complicated (not complex, but there's a lot that can go wrong, now and on futur update of it). Can you add a unit test for it ? I see there's a test that check that nothing change when the user is there, but it's different than "we get the correct values for error message", and it's all the ops will see. 

I think you should make an utility method alike `setExistingUsers` but return the non updated ones, or change signature of `setExistingUsers` to do that, and then check that you have the correct returned non updated user.

Reply:
:+1: That is right, I changed the `setExistingUsers` to return the non updated users, also previously it logged both conflicting users : "AlIcE" and "alice" instead of only the one which is passed for the update ("AlIcE" in the unit test)